### PR TITLE
Fix searchbar methods next

### PR DESCRIPTION
**Pull request by xavier-villelegier**
## Breaking changes in SearchBar

- `clearText()` now becomes `clear()` to be consistent with React Native `TextInput`
- `onClearText` now becomes `onClear`

### CODE REVIEW

File: src/index.d.ts [hunk 1]
```diff
@@ -1580,7 +1580,7 @@ export interface SearchBarPropsDefault extends TextInputProperties {
   /**
      * Method fired when text is cleared via the clear button
      */
-  onClearText?(): void;
+  onClear?(): void;
```
Comment (iRoachie) [inline src/index.d.ts]: YASSS he is updating TYPESCRIPT *does dance*

**Comment by xavier-villelegier** [comment]
@iRoachie The new `defaultProps` seem to affect the coverage, but maybe we don't care

**Comment by iRoachie** [comment]
Does clicking on the change for you show how it affects the coverage?

**Comment by iRoachie** [comment]
Needs rebase!  But you know this :P

**Comment by Monte9** [comment]
@xavier-villelegier can you make sure that this PR includes all the changes made in this PR: https://github.com/react-native-training/react-native-elements/pull/938

Cause #938 looked pretty useful and was closed in favor of your PR. So I wanna make sure we are not losing any useful work. Thanks

File: docs/searchbar.md [hunk 2]
```diff
@@ -83,16 +83,17 @@ import { SearchBar } from 'react-native-elements'
...
 | clearIcon | { color: '#86939e', name: 'close' } | object {name (string), color (string), style (object)} | specify color, styling, or another [Material Icon Name](https://design.google.com/icons/) (Note: pressing on this icon clears text inside the searchbar) |
 
 ##### Interaction methods
 | method | description |
 | ---- | ---- |
 | focus | call focus on the textinput ([example](#calling)) |
 | blur | call blur on the textinput ([example](#calling)) |
-| clearText | call clear on the textinput ([example](#calling))|
+| clear | call clear on the textinput ([example](#calling))|
+| cancel | **(Android and iOS SearchBars only)** call cancel on the SearchBar (left arrow on Android, Cancel button on iOS) ([example](#calling))|
```
Comment (iRoachie) [inline docs/searchbar.md]: Should maybe explain what cancelling does *e.g removes focus, hides keyboard*

**Pull request merged**